### ISSUE TITLE
McSPI bus testing 

### ISSUE THREAD

**Issue by networkimprov** [networkimprov/arch-packages#49]
Pls add something here about how this would be tested :-)

**Comment by mranostay** [comment]
This is really easy unless you want to test an actual device. 
Just using the spidev_test tool and white wire loopback (MOSI to MISO) would be enough to verify if data is passing.

@tmlind thoughts?

@networkimprov  You happen to have the part number for the female connection version of the debug HW row header?

**Comment by networkimprov** [comment]
See "mating products" column on https://www.harwin.com/products/M50-3600542/

**Comment by tmlind** [comment]
Yeh makes sense to me, we just want to know we've connected the right pads

**Comment by mranostay** [comment]
@tmlind requested a few samples of that part from Harwin.. should be here in a few days

**Comment by mranostay** [comment]
BTW silkscreen doesn't have pin 1 marked on that header.

**Comment by mranostay** [comment]
Confirmed SPI loopbacks on the header with MISO connected to MOSI... didn't validate chip selects with logic analyzer.

https://github.com/networkimprov/linux/tree/anvl-v4.7-spi-validation

```
[root@anvl mranostay]# modprobe spidev
[root@anvl mranostay]# dd if=/dev/urandom of=test.bin count=1
1+0 records in
1+0 records out
512 bytes copied, 0.00338745 s, 151 kB/s
[root@anvl mranostay]# ./spidev_test -i test.bin -o results.bin 
spi mode: 0x0
bits per word: 8
max speed: 500000 Hz (500 KHz)
[root@anvl mranostay]# md5sum test.bin results.bin 
6a762cdb0cebd91fa8bb00904ea95099  test.bin
6a762cdb0cebd91fa8bb00904ea95099  results.bin`
```

**Comment by networkimprov** [comment]
Re "do not upstream" why so?

**Comment by mranostay** [comment]
Spidev isn't a physical device but a test driver, and I know for a fact it will not be accepted since there is prior attempts of people trying.

**Comment by networkimprov** [comment]
OK, dropped Tony a note requesting review. Then I'll pull the safe commit to anvl-v4.7.

**Comment by tmlind** [comment]
Yeah looks OK to me, we can test with a real device later on then if spidev worked

**Comment by mranostay** [comment]
Also don't we can upstream the spi1 pins stuff either since mainline likes
to keep unneeded stuff out of the tree. And we have one CS we can't use
anyway

**Comment by networkimprov** [comment]
Tony asked on Code Review email thread about input enable requirement.

**Comment by tmlind** [comment]
also we don't want to configure unconnected pins for pm or at least need to measure the extra consumption :)

**Comment by mranostay** [comment]
On Fri, Sep 30, 2016 at 9:12 AM, Tony Lindgren <notifications@github.com>
wrote:

> also we don't want to configure unconnected pins for pm or at least need
> to measure the extra consumption :)
>
Well in theory it should be low since it is a driven bus. But agreed :)
